packages/@lwc/synthetic-shadow/src/3rdparty/polymer/path-composer.ts

```diff
@@ -17,6 +17,7 @@ subject to an additional IP rights grant found at http://polymer.github.io/PATEN
 
 import { isNull } from '@lwc/shared';
 import { getOwnerDocument } from '../../shared/utils';
+import { SyntheticShadowRoot } from '../../faux-shadow/shadow-root';
```

Comment:
I think this circular dep is trouble, can we have another kind of check here?

Reply:
Using SyntheticShadowRoot here is redundant. By the time pathComposer is run, the shadow-root polyfill would have already patched global 'ShadowRoot' property on window to be SyntheticShadowRoot.